src/core/task/index.ts

```diff
@@ -1915,6 +1936,13 @@ export class Task {
 								}
 								// now execute the tool like normal
 								const content = await extractTextFromFile(absolutePath)
+
+								// Setup file watcher to detect future changes to this file
+								await this.fileContextTracker.setupFileWatcher(relPath)
```

Comment:
do we need to call `setupFileWatcher` each time like this since we already do this in `trackFileContext `?